packages/react/src/components/Popover/Popover.tsx

```diff
@@ -53,57 +101,87 @@ interface PopoverProps extends DOMProps, StyleProps {
   shouldCloseOnInteractOutside?(element: HTMLElement): boolean;
 }
 
-const Popover = React.forwardRef<PopoverElement, PopoverProps>((props, forwardedRef) => {
+const Popover: React.FC<PopoverProps> = props => {
   const {
     children,
-    className: classNameProp,
+    className,
     css,
     isDismissable = true,
     isKeyboardDismissDisabled = false,
     isNonModal,
     isOpen,
+    offset = 4,
     onClose,
+    onEntered,
+    onExited,
+    overlayProps,
+    overlayRef,
+    placement = 'bottom',
+    scrollRef,
+    shouldFlip = true,
     shouldCloseOnBlur = false,
     shouldCloseOnInteractOutside,
-    ...other
+    triggerRef,
   } = props;
 
-  const popoverRef = React.useRef<HTMLDivElement>(null);
+  const [exited, setExited] = React.useState(isOpen);
 
-  const { dialogProps } = useDialog({ role: 'dialog' }, popoverRef);
-  const { overlayProps } = useOverlay(
-    {
-      isOpen,
-      isDismissable: isDismissable && isOpen,
-      isKeyboardDismissDisabled,
-      onClose,
-      shouldCloseOnBlur,
-      shouldCloseOnInteractOutside,
-    },
-    popoverRef,
-  );
-  const { modalProps } = useModal({ isDisabled: isNonModal });
+  const { overlayProps: positionProps } = useOverlayPosition({
+    offset,
+    isOpen,
+    overlayRef,
+    placement,
+    scrollRef,
+    shouldFlip,
+    targetRef: triggerRef,
+  });
+
+  const handleClose = React.useCallback(() => {
+    onClose?.();
+  }, [onClose]);
+
+  const handleEntered = React.useCallback(() => {
+    setExited(false);
+
+    onEntered?.();
+  }, [onEntered]);
+
+  const handleExited = React.useCallback(() => {
+    setExited(true);
 
-  const { className } = useStyles({ css });
+    onExited?.();
+  }, [onExited]);
 
-  if (!isOpen) return null;
+  const mounted = !!isOpen || !exited;
 
   return (
-    <OverlayContainer>
-      <FocusScope restoreFocus>
-        <div
-          {...mergeProps(other, overlayProps, modalProps, dialogProps)}
-          className={cx(className, classNameProp, 'manifest-popover')}
-          ref={mergeRefs(popoverRef, forwardedRef)}
-        >
-          <DismissButton onDismiss={onClose} />
-          {children}
-          <DismissButton onDismiss={onClose} />
-        </div>
-      </FocusScope>
-    </OverlayContainer>
+    <PopoverContext.Provider
+      value={{
+        isDismissable,
+        isKeyboardDismissDisabled,
+        isNonModal,
+        isOpen,
+        onClose: handleClose,
+        onEntered: handleEntered,
+        onExited: handleExited,
+        overlayProps,
+        overlayRef,
+        placement,
+        positionProps,
+        shouldCloseOnBlur,
+        shouldCloseOnInteractOutside,
+      }}
+    >
+      {mounted && (
```

Comment:
OK here is where the fix comes in. I needed to ensure the hooks in `PopoverContent` gets called after the `OverlayContainer` to preserve the call cycle of the hooks.